nominations/models.py

```diff
@@ -355,6 +470,76 @@ def save(self, *args, **kwargs):
         if self.vol_notes and not self.staff_notes:
             self.staff_notes = self.vol_notes
 
+    def create_related_objects(self):
+        """Create related nomination and questionnaire for application."""
+        if not self.nomination:
+            self.nomination = Nomination.objects.create()
+
+        if not self.questionnaire:
+            self.questionnaire = Questionnaire.objects.create()
+
+    def generate_application_status(self):
+        """Returns a generated status based on completion of various items."""
+
+        if self.nomination.status == 'incomplete':
+            if self.questionnaire.status == 'complete':
+                status = 'needs-group-form'
+            else:
+                status = 'needs-group-form-and-questionnaire'
+        else:
+            # nomination complete
+            if self.questionnaire.status == 'incomplete':
+                # needs questionaire
+                status = 'needs-questionnaire'
+            else:
+                # questionnaire complete
+                status = 'incomplete'
+        return status
+
+    def is_editable(self):
+        """Returns whether the user can edit this application."""
```

Comment:
What does it mean `user can edit this application`? Seems like data can be edited outside of these statuses. Which user are we talking about?

Reply:
Front-end users, i.e. groups. Will change wording.